lib/routes.js

```diff
@@ -21,58 +55,64 @@ var setup = function (app, router) {
-    res.status(200).json({ "value": "Stopping application"})
-    process.exit();
+    res.status(200).json({ "value": "Debug Mode On" })
   });
 
   router.get('/read/:ip/:port/:unitId/:regType/:startAddress/:quantity', (req, res) => {
-    // Example
-    // http://localhost:20900/read/192.168.255.1/502/1/holding/4110/1/float?offset=111&multiplier=2
-    logger.info("🛎️  Route /read/... " + JSON.stringify(req.params))
-
+    logger.info("🛎️  Route /read Queued")
     logger.debug("🧩 Incoming Params: " + JSON.stringify(req.params));
     logger.debug("🔑 Incoming keys: " + JSON.stringify(req.query));
 
-    let reading = null
-
-    // Conversion Values
-    req.query.conversion = (req.query.hasOwnProperty('conversion')) ? req.query.conversion : "raw"
-    req.query.endianness = (req.query.hasOwnProperty('endianness')) ? req.query.endianness : "little"
-    req.query.debug = (req.query.hasOwnProperty('debug')) ? req.query.debug : false
-
-
-    // Reading
-    modbus.readRegister(
-      req.params.ip,
-      req.params.port,
-      req.params.unitId,
-      req.params.regType,
-      req.params.startAddress,
-      req.params.quantity,
-      req.query.conversion,
-      req.query.endianness
-    ).then((reading) => {
-      if (isNaN(reading) || reading == null) {
-        res.status(400).json({ "params": req.params, "reading": String(reading), "error": "NaN" })
-      } else {
-        // Scaling
-        req.query.multiplier = (req.query.hasOwnProperty('multiplier')) ? req.query.multiplier : 1
-        req.query.offset = (req.query.hasOwnProperty('offset')) ? req.query.offset : 0
-        const transformedReading = transformReading(reading, req.query.offset, req.query.multiplier)
-
-        // Respond
-        if (req.query.debug) {
-          res.status(200).json({ "params": req.params, "query": req.query, "reading": reading, "transformedReading": transformedReading })
+    // Add item to queue with callback
+    q.push({ name: 'Modbus Reading' }, function (err) {
+      if (err) {
+        // Handle error
+        console.error(err);
+        return res.status(500).json({ error: 'Internal Server Error' });
+      }
+
+      let reading = null
+
+      // Conversion Values
+      req.query.conversion = (req.query.hasOwnProperty('conversion')) ? req.query.conversion : "interger"
+      req.query.endianness = (req.query.hasOwnProperty('endianness')) ? req.query.endianness : "big"
+      req.query.debug = (req.query.hasOwnProperty('debug')) ? req.query.debug : false
+
+      // Reading
+      modbus.readRegister(
+        req.params.ip,
+        req.params.port,
+        req.params.unitId,
+        req.params.regType,
+        req.params.startAddress,
+        req.params.quantity,
+        req.query.conversion,
+        req.query.endianness
+      ).then((reading) => {
+        if (isNaN(reading) || reading == null) {
+          return res.status(400).json({ "params": req.params, "reading": String(reading), "error": "NaN" })
```

Comment:
400 is the wrong error here; this is not a problem with the client's request. If this means the register you tried to read doesn't exist this should be 404 Not Found; if this is more of a transient situation then 502 Bad Gateway is the right code.

Reply:
Good point